src/miner.cpp

```diff
@@ -113,6 +114,22 @@ CBlockTemplate* Mining::CreateNewBlock(const CChainParams& chainparams) const
     pblocktemplate->vTxFees.push_back(-1); // updated at end
     pblocktemplate->vTxSigOps.push_back(-1); // updated at end
 
+    // Largest block you're willing to create:
+    // Miners can adjust downwards if they wish to throttle their blocks, for instance, to work around
+    // high orphan rates or other scaling problems.
+    uint32_t nBlockMaxSize = std::max<uint32_t>(1000, GetArg("-blockmaxsize", DEFAULT_BLOCK_MAX_SIZE));
+    if (pblock->nTime < sizeForkTime.load()) { // before the protocol upgrade, limit size.
+        nBlockMaxSize = std::min<uint32_t>(nBlockMaxSize, MAX_BLOCK_SIZE);
+    }
```

Comment:
I prefer curly brackets but for consistency should this if have them removed like all the others?